core/src/main/java/io/questdb/std/DirectLongList.java

```diff
@@ -195,17 +194,12 @@ private void setCapacityBytes(long capacity) {
             }
             final long oldCapacity = this.capacity;
             final long oldSize = this.pos - this.address;
-            try {
-                long address = Unsafe.realloc(this.address, oldCapacity, capacity, memoryTag);
-                this.capacity = capacity;
-                this.address = address;
-                this.limit = address + capacity;
-                this.pos = Math.min(this.limit, address + oldSize);
-                LOG.debug().$("resized [old=").$(oldCapacity).$(", new=").$(this.capacity).$(']').$();
-            } catch (Throwable t) {
-                close();
-                throw t;
-            }
+            final long address = Unsafe.realloc(this.address, oldCapacity, capacity, memoryTag);
```

Comment:
Can you explain the logic behind this change? `close()` on exception here doesn't seem to carry a risk of failure.

Reply:
Due to this behavior we had direct lists in `PageFrameReduceTask` being closed implicitly. The current assumption is that we keep these lists pre-allocated until server shutdown, so keeping the list allocated on failed realloc keeps that contract in force. Any code that wants to close an initialized DirectInt/LongList on OOM error can do it explicitly since it owns the object.